content/posts/the-three-rs-rust-repl-and-runners/index.mdx

```diff
@@ -0,0 +1,234 @@
+---
+title: 'The Three "R"s: Rust, REPL, Runner'
+date: 2021-01-22
+---
+
+REPLs and runners can be an easier and quick way to iterate and execute code. Apply the same Rust example with a Cargo package and different Rust REPLS & Runners to see the differences between each tool.
+
+## Intro
+tl;dr [Jump to the code](#cargo-project)
+
+When evaluating a new libraries I often find myself going through the same steps:
+1. Do I have a pre-existing applicable project in source control?
+	1. Yes, create a branch
+	2. No, create a new Cargo package
+2. Add the dependency to `Cargo.toml`
+3. Change some relevant code to try something out about the library
+4. Build
+5. Run
+6. Do I have everything I need to know to make a decision about the library
+	1. Yes, go do what I need to do
+	2. No, go back to Step 3
+
+These steps can grow tedious, especially when making a new cargo package, and the iterations can be shortened by more work up front looking at documentation, source, and examples. But sometimes you just want to experiment in a quick and efficient way. That is when REPLs and runners can prove to be helpful.
+
+For the uninitiated, a REPL is typically an interactive shell that takes user input, evaluates said input, and returns the result.  Assuming I can access the required dependencies, I should be able to take the contents of the `fn main()` above and enter each expressions in line-by-line in a REPL.
+
+I did some digging to find an official Rust REPL and there is an open issue to officially support [implement a proper REPL · Issue #655 · rust-lang/rfcs · GitHub](https://github.com/rust-lang/rfcs/issues/655).
+
+While there isn’t an official REPL, there is at least one active Rust REPL:
+* [Evcxr REPL](https://github.com/google/evcxr/blob/master/evcxr_repl/)
+
+A Rust runner allows you to run (well, compile, link, _and_ run) Rust without the _bloat_ of Cargo. Similar to the REPLs, there isn’t an official Rust runner, besides Cargo, and there are a few options:
```

Comment:
Is cargo really bloat if it's part of the a standard Rust install so you can rely on anyone coding in Rust having it? Or do you mean that without the hassle of creating a full cargo project and the structure it entails. If it's the latter then I'd suggest rewording to make it clearer.

Reply:
It is certainly the latter. Seems I may have left a bit too much masked behind `_bloat_`. I've changed the language to better reflect what I meant.

```
--- a/content/posts/the-three-rs-rust-repl-and-runners/index.mdx
+++ b/content/posts/the-three-rs-rust-repl-and-runners/index.mdx
@@ -29,7 +29,7 @@ I did some digging to find an official Rust REPL and there is an open issue to o
 While there isn’t an official REPL, there is at least one active Rust REPL:
 * [Evcxr REPL](https://github.com/google/evcxr/blob/master/evcxr_repl/)
 
-A Rust runner allows you to run (well, compile, link, _and_ run) Rust without the _bloat_ of Cargo. Similar to the REPLs, there isn’t an official Rust runner, besides Cargo, and there are a few options:
+A Rust runner allows you to run (well, compile, link, _and_ run) Rust code without the overhead (and benefits) of creating a Cargo package. Similar to the REPLs, there isn’t an official Rust runner, besides Cargo, and there are a few options:
 * [runner](https://github.com/stevedonovan/runner)
 * [cargo-eval](https://github.com/reitermarkus/cargo-eval)
 ```